src/main/java/com/example/library/repository/BookRepositoryImpl.java

```diff
@@ -1,54 +1,48 @@
 package com.example.library.repository;
 
 import com.example.library.model.Book;
-import jakarta.persistence.criteria.CriteriaQuery;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.EntityManagerFactory;
+import jakarta.persistence.EntityTransaction;
 import java.util.List;
-import org.hibernate.Session;
-import org.hibernate.SessionFactory;
-import org.hibernate.Transaction;
-import org.springframework.beans.factory.annotation.Autowired;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Repository;
 
+@RequiredArgsConstructor

```

Comment:
Why do you modify these methods? You will remove them in the next HW so let’s skip it for now.